### ISSUE TITLE
[GEN2] "500 runtime error: invalid memory address or nil pointer dereference" when not enough memory

### ISSUE THREAD

**Issue by ndbroadbent** [convox/rack#2537]
~I was reading #2422 and saw that multiple domains could be specified as an array.~

I got the following error:

```
Starting build... ERROR: response status: 500 runtime error: invalid memory address or nil pointer dereference
```

After looking at the rack logs, I figured out that it was because I was requesting too much memory for the build.

Error stopped happening when I ran `convox rack params set BuildMemory=1024`.

**Comment by ndbroadbent** [comment]
Sorry that was unrelated, I looked at the logs with `convox rack logs`, and I think I just didn't have enough memory to start the build:


```
2018-03-09T17:53:53Z service/web:20180308185456/29ba273cc81b ns=provider.aws at=BuildCreate app="myapp" method="tgz" url="object://myapp/tmp/0a8b3d249444916eb80ae523830411" state=error error="not enough memory available to start process" location="/go/src/github.com/convox/rack/provider/aws/builds.go:71" elapsed=416.931
2018-03-09T17:53:53Z service/web:20180308185456/29ba273cc81b ns=kernel at=panic state=error id=2008019434 message="runtime error: invalid memory address or nil pointer dereference"
2018-03-09T17:53:53Z service/web:20180308185456/29ba273cc81b ns=kernel at=panic state=error id=2008019434 line=1 trace="goroutine 4640 [running]:"
2018-03-09T17:53:53Z service/web:20180308185456/29ba273cc81b ns=kernel at=panic state=error id=2008019434 line=2 trace="runtime/debug.Stack(0xc42052ca80, 0x1218ecf, 0x1c)"
2018-03-09T17:53:53Z service/web:20180308185456/29ba273cc81b ns=kernel at=panic state=error id=2008019434 line=3 trace="\t/usr/local/go/src/runtime/debug/stack.go:24 +0xa7"
2018-03-09T17:53:53Z service/web:20180308185456/29ba273cc81b ns=kernel at=panic state=error id=2008019434 line=4 trace="github.com/convox/rack/vendor/github.com/convox/logger.(*Logger).ErrorBacktrace(0xc42052ca80, 0x1a85d60, 0x1cb1a00)"
2018-03-09T17:53:53Z service/web:20180308185456/29ba273cc81b ns=kernel at=panic state=error id=2008019434 line=5 trace="\t/go/src/github.com/convox/rack/vendor/github.com/convox/logger/logger.go:80 +0x123"
2018-03-09T17:53:53Z service/web:20180308185456/29ba273cc81b ns=kernel at=panic state=error id=2008019434 line=6 trace="github.com/convox/rack/helpers.Error(0xc42052ca80, 0x1a85d60, 0x1cb1a00)"
2018-03-09T17:53:53Z service/web:20180308185456/29ba273cc81b ns=kernel at=panic state=error id=2008019434 line=7 trace="\t/go/src/github.com/convox/rack/helpers/helpers.go:38 +0x4c5"
2018-03-09T17:53:53Z service/web:20180308185456/29ba273cc81b ns=kernel at=panic state=error id=2008019434 line=8 trace="github.com/convox/rack/server.recovery.func1(0x1a85d60, 0x1cb1a00)"
2018-03-09T17:53:53Z service/web:20180308185456/29ba273cc81b ns=kernel at=panic state=error id=2008019434 line=9 trace="\t/go/src/github.com/convox/rack/server/server.go:72 +0xc9"
2018-03-09T17:53:53Z service/web:20180308185456/29ba273cc81b ns=kernel at=panic state=error id=2008019434 line=10 trace="github.com/convox/rack/server.recoverWith(0xc4207dbc08)"
2018-03-09T17:53:53Z service/web:20180308185456/29ba273cc81b ns=kernel at=panic state=error id=2008019434 line=11 trace="\t/go/src/github.com/convox/rack/server/server.go:87 +0x88"
2018-03-09T17:53:53Z service/web:20180308185456/29ba273cc81b ns=kernel at=panic state=error id=2008019434 line=12 trace="panic(0x1027440, 0x1cb1a00)"
2018-03-09T17:53:53Z service/web:20180308185456/29ba273cc81b ns=kernel at=panic state=error id=2008019434 line=13 trace="\t/usr/local/go/src/runtime/panic.go:491 +0x283"
2018-03-09T17:53:53Z service/web:20180308185456/29ba273cc81b ns=kernel at=panic state=error id=2008019434 line=14 trace="github.com/convox/rack/api/controllers.BuildCreate(0x1a895e0, 0xc42039e7a0, 0xc4200fdf00, 0xc4207a22c0)"
2018-03-09T17:53:53Z service/web:20180308185456/29ba273cc81b ns=kernel at=panic state=error id=2008019434 line=15 trace="\t/go/src/github.com/convox/rack/api/controllers/builds.go:65 +0x1361"
2018-03-09T17:53:53Z service/web:20180308185456/29ba273cc81b ns=provider.aws at=runBuild method="tgz" url="object://myapp/tmp/0a8b3d249444916eb80ae523830411" state=error error="not enough memory available to start process" location="/go/src/github.com/convox/rack/provider/aws/builds.go:941" elapsed=342.670
2018-03-09T17:53:53Z service/web:20180308185456/29ba273cc81b ns=kernel at=panic state=error id=2008019434 line=17 trace="\t/go/src/github.com/convox/rack/api/controllers/api.go:37 +0x10c"
2018-03-09T17:53:53Z service/web:20180308185456/29ba273cc81b ns=kernel at=panic state=error id=2008019434 line=18 trace="net/http.HandlerFunc.ServeHTTP(0xc420262680, 0x1a895e0, 0xc42039e7a0, 0xc4200fdf00)"
2018-03-09T17:53:53Z service/web:20180308185456/29ba273cc81b ns=kernel at=panic state=error id=2008019434 line=19 trace="\t/usr/local/go/src/net/http/server.go:1918 +0x44"
2018-03-09T17:53:53Z service/web:20180308185456/29ba273cc81b ns=kernel at=panic state=error id=2008019434 line=20 trace="github.com/convox/rack/vendor/github.com/gorilla/mux.(*Router).ServeHTTP(0xc420107080, 0x1a895e0, 0xc42039e7a0, 0xc4200fdf00)"
2018-03-09T17:53:53Z service/web:20180308185456/29ba273cc81b ns=kernel at=panic state=error id=2008019434 line=21 trace="\t/go/src/github.com/convox/rack/vendor/github.com/gorilla/mux/mux.go:150 +0xed"
2018-03-09T17:53:53Z service/web:20180308185456/29ba273cc81b ns=kernel at=panic state=error id=2008019434 line=22 trace="github.com/convox/rack/vendor/github.com/codegangsta/negroni.Wrap.func1(0x1a895e0, 0xc42039e7a0, 0xc4200fdd00, 0xc42039e7c0)"
2018-03-09T17:53:53Z service/web:20180308185456/29ba273cc81b ns=kernel at=panic state=error id=2008019434 line=23 trace="\t/go/src/github.com/convox/rack/vendor/github.com/codegangsta/negroni/negroni.go:46 +0x4d"
2018-03-09T17:53:53Z service/web:20180308185456/29ba273cc81b ns=kernel at=panic state=error id=2008019434 line=24 trace="github.com/convox/rack/vendor/github.com/codegangsta/negroni.HandlerFunc.ServeHTTP(0xc4203170c0, 0x1a895e0, 0xc42039e7a0, 0xc4200fdd00, 0xc42039e7c0)"
2018-03-09T17:53:53Z service/web:20180308185456/29ba273cc81b ns=kernel at=panic state=error id=2008019434 line=25 trace="\t/go/src/github.com/convox/rack/vendor/github.com/codegangsta/negroni/negroni.go:29 +0x4e"
2018-03-09T17:53:53Z service/web:20180308185456/29ba273cc81b ns=kernel at=panic state=error id=2008019434 line=26 trace="github.com/convox/rack/vendor/github.com/codegangsta/negroni.middleware.ServeHTTP(0x1a84e60, 0xc4203170c0, 0xc420317140, 0x1a895e0, 0xc42039e7a0, 0xc4200fdd00)"
2018-03-09T17:53:53Z service/web:20180308185456/29ba273cc81b ns=kernel at=panic state=error id=2008019434 line=27 trace="\t/go/src/github.com/convox/rack/vendor/github.com/codegangsta/negroni/negroni.go:38 +0xa5"
2018-03-09T17:53:53Z service/web:20180308185456/29ba273cc81b ns=kernel at=panic state=error id=2008019434 line=28 trace="github.com/convox/rack/vendor/github.com/codegangsta/negroni.(middleware).ServeHTTP-fm(0x1a895e0, 0xc42039e7a0, 0xc4200fdd00)"
2018-03-09T17:53:53Z service/web:20180308185456/29ba273cc81b ns=kernel at=panic state=error id=2008019434 line=29 trace="\t/go/src/github.com/convox/rack/vendor/github.com/codegangsta/negroni/negroni.go:38 +0x60"
2018-03-09T17:53:53Z service/web:20180308185456/29ba273cc81b ns=kernel at=panic state=error id=2008019434 line=30 trace="github.com/convox/rack/vendor/github.com/convox/nlogger.(*Nlogger).ServeHTTP(0xc420262120, 0x7f9e446c09f0, 0xc4200f0310, 0xc4200fdd00, 0xc42039e780)"
2018-03-09T17:53:53Z service/web:20180308185456/29ba273cc81b ns=kernel at=panic state=error id=2008019434 line=31 trace="\t/go/src/github.com/convox/rack/vendor/github.com/convox/nlogger/nlogger.go:56 +0xbb"
2018-03-09T17:53:53Z service/web:20180308185456/29ba273cc81b ns=kernel at=panic state=error id=2008019434 line=32 trace="github.com/convox/rack/vendor/github.com/codegangsta/negroni.middleware.ServeHTTP(0x1a82ba0, 0xc420262120, 0xc420317120, 0x7f9e446c09f0, 0xc4200f0310, 0xc4200fdd00)"
2018-03-09T17:53:53Z service/web:20180308185456/29ba273cc81b ns=kernel at=panic state=error id=2008019434 line=33 trace="\t/go/src/github.com/convox/rack/vendor/github.com/codegangsta/negroni/negroni.go:38 +0xa5"
2018-03-09T17:53:53Z service/web:20180308185456/29ba273cc81b ns=kernel at=panic state=error id=2008019434 line=34 trace="github.com/convox/rack/vendor/github.com/codegangsta/negroni.(middleware).ServeHTTP-fm(0x7f9e446c09f0, 0xc4200f0310, 0xc4200fdd00)"
2018-03-09T17:53:53Z service/web:20180308185456/29ba273cc81b ns=kernel at=panic state=error id=2008019434 line=35 trace="\t/go/src/github.com/convox/rack/vendor/github.com/codegangsta/negroni/negroni.go:38 +0x60"
2018-03-09T17:53:53Z service/web:20180308185456/29ba273cc81b ns=kernel at=panic state=error id=2008019434 line=36 trace="github.com/convox/rack/server.development(0x7f9e446c09f0, 0xc4200f0310, 0xc4200fdd00, 0xc42039e760)"
2018-03-09T17:53:53Z service/web:20180308185456/29ba273cc81b ns=kernel at=panic state=error id=2008019434 line=37 trace="\t/go/src/github.com/convox/rack/server/server.go:66 +0x7a"
2018-03-09T17:53:53Z service/web:20180308185456/29ba273cc81b ns=kernel at=panic state=error id=2008019434 line=38 trace="github.com/convox/rack/vendor/github.com/codegangsta/negroni.HandlerFunc.ServeHTTP(0x1233778, 0x7f9e446c09f0, 0xc4200f0310, 0xc4200fdd00, 0xc42039e760)"
2018-03-09T17:53:53Z service/web:20180308185456/29ba273cc81b ns=kernel at=panic state=error id=2008019434 line=39 trace="\t/go/src/github.com/convox/rack/vendor/github.com/codegangsta/negroni/negroni.go:29 +0x4e"
2018-03-09T17:53:53Z service/web:20180308185456/29ba273cc81b ns=kernel at=panic state=error id=2008019434 line=40 trace="github.com/convox/rack/vendor/github.com/codegangsta/negroni.middleware.ServeHTTP(0x1a84e60, 0x1233778, 0xc420317100, 0x7f9e446c09f0, 0xc4200f0310, 0xc4200fdd00)"
2018-03-09T17:53:53Z service/web:20180308185456/29ba273cc81b ns=kernel at=panic state=error id=2008019434 line=41 trace="\t/go/src/github.com/convox/rack/vendor/github.com/codegangsta/negroni/negroni.go:38 +0xa5"
2018-03-09T17:53:53Z service/web:20180308185456/29ba273cc81b ns=kernel at=panic state=error id=2008019434 line=42 trace="github.com/convox/rack/vendor/github.com/codegangsta/negroni.(middleware).ServeHTTP-fm(0x7f9e446c09f0, 0xc4200f0310, 0xc4200fdd00)"
2018-03-09T17:53:53Z service/web:20180308185456/29ba273cc81b ns=kernel at=panic state=error id=2008019434 line=43 trace="\t/go/src/github.com/convox/rack/vendor/github.com/codegangsta/negroni/negroni.go:38 +0x60"
2018-03-09T17:53:53Z service/web:20180308185456/29ba273cc81b ns=kernel at=panic state=error id=2008019434 line=46 trace="github.com/convox/rack/vendor/github.com/codegangsta/negroni.HandlerFunc.ServeHTTP(0x1233788, 0x7f9e446c09f0, 0xc4200f0310, 0xc4200fdd00, 0xc42039e740)"
2018-03-09T17:53:53Z service/web:20180308185456/29ba273cc81b ns=kernel at=panic state=error id=2008019434 line=45 trace="\t/go/src/github.com/convox/rack/server/server.go:76 +0x97"
2018-03-09T17:53:53Z service/web:20180308185456/29ba273cc81b ns=kernel at=panic state=error id=2008019434 line=44 trace="github.com/convox/rack/server.recovery(0x7f9e446c09f0, 0xc4200f0310, 0xc4200fdd00, 0xc42039e740)"
2018-03-09T17:53:53Z service/web:20180308185456/29ba273cc81b ns=kernel at=panic state=error id=2008019434 line=53 trace="\t/usr/local/go/src/net/http/server.go:2619 +0xb4"
2018-03-09T17:53:53Z service/web:20180308185456/29ba273cc81b ns=kernel at=panic state=error id=2008019434 line=48 trace="github.com/convox/rack/vendor/github.com/codegangsta/negroni.middleware.ServeHTTP(0x1a84e60, 0x1233788, 0xc4203170e0, 0x7f9e446c09f0, 0xc4200f0310, 0xc4200fdd00)"
2018-03-09T17:53:53Z service/web:20180308185456/29ba273cc81b ns=kernel at=panic state=error id=2008019434 line=49 trace="\t/go/src/github.com/convox/rack/vendor/github.com/codegangsta/negroni/negroni.go:38 +0xa5"
2018-03-09T17:53:53Z service/web:20180308185456/29ba273cc81b ns=kernel at=panic state=error id=2008019434 line=50 trace="github.com/convox/rack/vendor/github.com/codegangsta/negroni.(*Negroni).ServeHTTP(0xc4202351d0, 0x1a89b20, 0xc42054a2a0, 0xc4200fdd00)"
2018-03-09T17:53:53Z service/web:20180308185456/29ba273cc81b ns=kernel at=panic state=error id=2008019434 line=51 trace="\t/go/src/github.com/convox/rack/vendor/github.com/codegangsta/negroni/negroni.go:96 +0xf5"
2018-03-09T17:53:53Z service/web:20180308185456/29ba273cc81b ns=kernel at=panic state=error id=2008019434 line=52 trace="net/http.serverHandler.ServeHTTP(0xc4205c2000, 0x1a89b20, 0xc42054a2a0, 0xc4200fdd00)"
2018-03-09T17:53:53Z service/web:20180308185456/29ba273cc81b ns=kernel at=panic state=error id=2008019434 line=54 trace="net/http.(*conn).serve(0xc422dc2c80, 0x1a8a5e0, 0xc4207a2100)"
2018-03-09T17:53:53Z service/web:20180308185456/29ba273cc81b ns=kernel at=panic state=error id=2008019434 line=55 trace="\t/usr/local/go/src/net/http/server.go:1801 +0x71d"
2018-03-09T17:53:53Z service/web:20180308185456/29ba273cc81b ns=kernel at=panic state=error id=2008019434 line=56 trace="created by net/http.(*Server).Serve"
2018-03-09T17:53:53Z service/web:20180308185456/29ba273cc81b ns=kernel at=panic state=error id=2008019434 line=57 trace="\t/usr/local/go/src/net/http/server.go:2720 +0x288"
2018-03-09T17:53:53Z service/web:20180308185456/29ba273cc81b ns=kernel at=panic state=error id=2008019434 line=47 trace="\t/go/src/github.com/convox/rack/vendor/github.com/codegangsta/negroni/negroni.go:29 +0x4e"
```

**Comment by jontonsoup** [comment]
I am getting this too. From what I am seeing, it appears to be memory related

**Comment by ddollar** [comment]
Will be resolved by #2546
